### DISCUSSION TITLE
Smart Image Decomposition

### DISCUSSION THREAD

**Discussion by ironhacker** [ImageMagick/ImageMagick#7385] [Digital Image Processing]
On many Canon scanners I've used, creating a PDF often results in the image being broken into multiple component images. For instance, a text document with a blue signature might be saved with a CCITT G4 image for most of the page and a small indexed PNG for the signature. This approach is very space-efficient and maintains excellent visual quality. Can ImageMagick perform this type of analysis out of the box or with some effort? If not, are there any open-source or commercial software options available that can?  My objective is to batch convert scanned images to PDF with these optimizations.

**Comment by fmw42** [comment]
I don't think IM can do that.  IM will always rasterize any PDF when reading and when writing back to PDF. So any vector information will be converted to raster.  I doubt you want that as it will increase your file size.

**Reply by ironhacker** [reply]
No need to start with a PDF.  I'm starting with scanned images.  Mostly I want to separate an image into components, but it needs to be analyzed to identify which formats to use e.g. CCIT G4 for b/w, JPEG for photo, indexed PNG for non-photo color, etc.  A CCIT image with a small JPEG is much smaller than the entire image as JPEG.  Once decomposed, recombining them would be easy.  It's the analysis (coordinates and color profile) that I'm looking for.
